src/state/reducers/month-state/shifts-info.reducer.ts

```diff
@@ -26,6 +26,9 @@ export function scheduleShiftsInfoReducerF(name: string) {
       case createActionName(name, ScheduleActionType.UPDATE):
         const data = (action.payload as ScheduleDataModel)?.shifts;
         return { ...data };
+      case ScheduleActionType.DELETE_WORKER:
+        delete state[workerName];
+        return state;
```

Comment:
Lepiej robić spread, czyli 
`return {...state}`
Niektóre elementy ( wszystkie które używają `useSelector` nie podając w niego drugi argument, który właśnie odpowiada za to, żeby customowo porównywać obiekty ) sprawdzają czy stan się zmienił na podstawie referencji ( porównują referencję nowego obiektu i poprzedniego )
W tym przypadku obie referencje będą równe więc komponenty, które w taki sposób śledzą stan tego nie zauważą